### PR TITLE
add temp socket error handler in case error occur before accept  #407

### PR DESCRIPTION
**Pull request by panshengjie**

### CODE REVIEW

File: lib/WebSocketRequest.js [hunk 1]
```diff
@@ -502,12 +504,19 @@ WebSocketRequest.prototype.reject = function(status, reason, extraHeaders) {
     this.emit('requestRejected', this);
 };
 
+WebSocketRequest.prototype._handleSocketErrorBeforeAccept = function() {
+    //If error happen before accept, do nothing here. socket will close right after this.
+    //But we need this, or the app crash without socket error handler (if no default error handler)
+    //see https://github.com/theturtle32/WebSocket-Node/issues/407
```
Comment (ibc) [inline lib/WebSocketRequest.js:510]:
```suggestion
    // If error happen before accept, do nothing here. socket will close right after this.
    // But we need this, or the app crash without socket error handler (if no default error handler),
    // see https://github.com/theturtle32/WebSocket-Node/issues/407
```

**Review by ibc — APPROVED** [review]
just a cosmetic change requested, otherwise LGTM

**Comment by ibc** [comment]
Have you tested this PR?

**Comment by panshengjie** [comment]
not test yet



---Original---

**Comment by ibc** [comment]
ok, please confirm when tested.

**Comment by ASBaj** [comment]
Hi, was curious if there was a reason this hasn't been merged into the main branch yet. I too have been running into the ECONNRESET error, and the changes here do appear to fix the issue.

**Comment by panshengjie** [comment]
actually it's ok far early , but I forgot this

**Pull request closed without merging**